openedx/core/djangoapps/discussions/docs/decisions/0006-discussions-api-fetch-inactive-providers.rst

```diff
@@ -0,0 +1,150 @@
+without needing to switch to it first.
+
+Alternatives Considered
+-----------------------
+
+Another way around the second issue above is to save the provider change
+first, fetch the configuration data and then proceed with the second
+configuration step.
+
+However this can be very disruptive since changing the provider first will
+trigger a course republish and a lot of background machinery to build
+discussion topics, and will change the live experience using old settings that
+the course author might have wanted to change before publishing.
+
+Decision
+--------
+
+We can split the existing configuration API into two parts. The first part will
+be a read-only API for listing provider details. No data in this API can be
+modified by the user.
+
+The other API will return all the writable configuration for the providers.
+It will support updating the settings of the current provider as it does
+now, but will also support returning the configuration/settings of other
+providers by taking a ``provider_id`` query parameter.
+
+
+Provider Endpoint
+~~~~~~~~~~~~~~~~~
+
+``/api/discussions/v0/course/{course_id}/providers``
+
+This will return the list of available providers and their data. This will
+return the following pieces of information:
+
+- ``active``: the active provider
+- ``features``: feature support levels
+- ``available``: provider listing
+
+**Sample response:**
+
+.. code-block:: json
+
+    {
+        "active": "legacy",
+        "features": [
+            {
+                "id": "primary-discussion-app-experience",
+                "feature_support_type": "basic"
+            },
+            …
+        ],
+        "available": {
+            "legacy": {
+                "features": [ … ],
+                "supports_lti": false,
+                "external_links": { … },
+                "messages": [ … ],
+                "has_full_support": true
+            },
+            …
+        }
+    }
+
+
+Settings Endpoint
+~~~~~~~~~~~~~~~~~
+
+``/api/discussions/v0/course/{course_id}/settings``
+
+This is the read-write settings API that can take an optional `provider_id`
+argument in which case returns data for that provider.
+
+This will return the following information which is common for all providers:
+
+- enabled
+- provider_type
+- enable_in_context
+- enable_graded_units
+- unit_level_visibility
+
+The following information will vary based on provider specified:
+
+- lti_configuration: only returned for the active provider.
+- plugin_configuration: plugin configuration for the specified provider.
+
+Changing settings for an inactive provider is out of scope here.
+
+**Sample response:**
+
+.. code-block:: json
+
+    {
+      "enabled": true,
+      "provider_type": "legacy",
+      "enable_in_context": true,
+      "enable_graded_units": true,
+      "unit_level_visibility": false,
+      "lti_configuration": {},
+      "plugin_configuration": { … }
+    }
```

Comment:
@xitij2000 Maybe its also helpful to conclude the ADR stating how this selected approach will resolve the issue of providers switching?

Reply:
Done.